packages/core/src/routes/account/third-party-tokens.ts

```diff
@@ -3,31 +3,49 @@ import {
   getThirdPartyAccessTokenResponseGuard,
   type GetThirdPartyAccessTokenResponse,
   type SocialTokenSetSecret,
+  type TokenSetMetadata,
 } from '@logto/schemas';
 import { conditional } from '@silverhand/essentials';
 import { z } from 'zod';
 
 import RequestError from '#src/errors/RequestError/index.js';
 import koaGuard from '#src/middleware/koa-guard.js';
+import type Libraries from '#src/tenants/Libraries.js';
 import type Queries from '#src/tenants/Queries.js';
 import { decryptTokens } from '#src/utils/secret-encryption.js';
 
 import { type UserRouter, type RouterInitArgs } from '../types.js';
 
 import { accountApiPrefix } from './constants.js';
 
+const isSocialTokenSetSecret = (
+  tokenSetSecret: SocialTokenSetSecret | EnterpriseSsoTokenSetSecret
+): tokenSetSecret is SocialTokenSetSecret =>
+  'connectorId' in tokenSetSecret && 'target' in tokenSetSecret;
+
+const formatTokenResponse = (
+  access_token: string,
+  metadata: TokenSetMetadata,
+  now = Math.floor(Date.now() / 1000)
```

Comment:
is there any case that we will specify value for `now` input param?

Reply:
Yes, we will validate the token expiration status ahead. So the `now` value may be generated ahead of this util method. In this case, we would like to use the same value instead of generating a new one.